pkg/types/powervs/validation/machinepool_test.go

```diff
@@ -0,0 +1,156 @@
+package validation
+
+import (
+	"testing"
+
+	"github.com/openshift/installer/pkg/types/powervs"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+func TestValidateMachinePool(t *testing.T) {
+	cases := []struct {
+		name     string
+		pool     *powervs.MachinePool
+		expected string
+	}{
+		{
+			name: "empty",
+			pool: &powervs.MachinePool{},
+		},
+		{
+			name: "valid serviceinstance",
+			pool: &powervs.MachinePool{
+				ServiceInstance: "637efe83-0078-46c1-a15d-7d0d3facd651",
+			},
+		},
+		{
+			name: "invalid serviceinstance",
+			pool: &powervs.MachinePool{
+				ServiceInstance: "powervs-service-instance",
+			},
+			expected: `^test-path\.serviceinstance: Invalid value: "powervs-service-instance": Service Instance is provided is not a UUID$`,
+		},
+		{
+			name: "valid name",
+			pool: &powervs.MachinePool{
+				Name: "powervs-test",
+			},
+		},
+		{
+			name: "invalid name",
+			pool: &powervs.MachinePool{
+				Name: "powervs-test!",
+			},
+			expected: `^test-path\.name: Invalid value: "powervs-test!": Only letters (no accents), numbers, underscores and dashes are allowed$`,
+		},
+		{
+			name: "valid volumeIDs",
+			pool: &powervs.MachinePool{
+				VolumeIDs: []string{"c8b709c4-93f1-499e-915e-0820bcc51406", "587c5788-107f-4351-aabc-1652c54c4491"},
```

Comment:
can you also add a check to make sure that an empty list behaves correctly?

Reply:
Done!

empty list is considered a valid set of volumeIDs, correct?

Reply:
@clnperez I think this is covered by the "empty" test, but let me know if I'm wrong.

Reply:
you've got an empty machine pool test, but i don't see a machinepool test where the volumeids list is empty